src/shape.cpp

```diff
@@ -663,10 +676,15 @@ void migraphx_from_value(const value& v, shape& s)
             auto v_dd = v.at("dynamic_dimensions");
             std::vector<shape::dynamic_dimension> dyn_dims(v.at("dynamic_dimensions").size());
             std::transform(v_dd.begin(), v_dd.end(), dyn_dims.begin(), [](migraphx::value x) {
-                auto x_min = x.at("min").template to<size_t>();
-                auto x_max = x.at("max").template to<size_t>();
-                auto x_opt = x.at("opt").template to<size_t>();
-                return shape::dynamic_dimension{x_min, x_max, x_opt};
+                auto x_min  = x.at("min").template to<size_t>();
+                auto x_max  = x.at("max").template to<size_t>();
+                auto v_opts = x.at("opts");
```

Comment:
I think you can write `auto set_x_opts = from_value<std::set<std::size_t>>(x.at("opts"));`